scripts/reduction_gui/reduction/diffraction/diffraction_adv_setup_script.py

```diff
@@ -291,9 +291,11 @@ def from_xml(self, xml_str):
             if string_tmp:
                 items = string_tmp.replace("{", "").replace("}", "").split(",")
                 keys_tmp = [item.split(":")[0].replace("'", "") for item in items]
-                vals_tmp = [float(item.split(":")[1].replace("'", "")) for item in items]
-                for count, value in enumerate(keys_tmp):
-                    self.samplegeometry[value] = vals_tmp[count]
+                vals_tmp = [item.split(":")[1].replace("'", "").strip() for item in items]
+                if vals_tmp.count("") == 0:
```

Comment:
When height information is not provided in POWGEN GUI, the sample geometry entry read in from the saved XML file will be something like this,

```json
"samplegeometry": {"Height": " "}
```

when the original code was processing something like this, it would try to `float` the value corresponding to the entry of `Height` and therefore failed. What I was doing here is, given all the values in the dict corresponding to `samplegeometry`, I will first strip spaces for all values (in this case, we only have one value which is corresponding to `Height`). Then by `vals_tmp.count("")`, I am counting how many empty values we have. If the number of empty values is 0, that means we have valid values for all entries and then we can continue to the `for` loop as in what follows.